flow/flow.go

```diff
@@ -363,13 +367,68 @@ func (f *Flow) Init(key string, now int64, packet *gopacket.Packet, length int64
 	f.UpdateUUID(key, L2ID, L3ID)
 }
 
-// Update a flow metrics
+// Update a flow metrics and latency
 func (f *Flow) Update(now int64, packet *gopacket.Packet, length int64) {
 	f.Last = now
 
 	if updated := f.updateMetricsWithLinkLayer(packet, length); !updated {
 		f.updateMetricsWithNetworkLayer(packet)
 	}
+
+	f.updateLatency(packet)
+}
+
+func (f *Flow) updateLatency(packet *gopacket.Packet) {
+	if f.Link != nil && f.Link.latencySet == false {
+		ethernetLayer := (*packet).Layer(layers.LayerTypeEthernet)
+		ethernetPacket, ok := ethernetLayer.(*layers.Ethernet)
+		if ok && (f.Link.A == ethernetPacket.DstMAC.String()) {
+			f.Link.Latency = (*packet).Metadata().Timestamp.UnixNano() - f.Link.Latency
+			f.Link.latencySet = true
+		}
+	}
```

Comment:
Yes the patch assume Request/Response of the upper layer
In the scenario of multiple syn send by to open the connection, what is important is to capture the 1st syn and the first syn/ack, or rst/fin what ever scenario, take in account the syn packets (2nd and ...) make no much sense in latency / RTT calculation. we can call that RTT Network

I was planned in a futur patch to add the Application layer, in this way we could calculate the RTT Application, in case of http for example : the time between the first http request and 1st http response, it could be ICMP, ... 

An good measure is network health ratio that give you a good value of the current connection health
NHR = RTTApplication / RTTNetwork

I seen your patch but I think you try to address too many use-cases at once, like synflood, port closed, fake checksum ... I would prefer to address a problem by integrating an external flow enhancer (another patch under way) with an IDS like [prelude](https://www.prelude-siem.org/) or other existing tools, as there are many tcp attack or other attack already implemented and will not have a performance impact on Skydive, so only the connection problem/attack/... would be logged in the Sydive database
Meanwhile I could understand that you need the 1st rawpacket on each way, aim to debug a connection problem